File: frontend/src/api/batch.ts
/**
 * 批次管理 API
 */
import request from './request'

export interface Batch {
  id: number
  name: string
  description?: string
  rule_ids: number[]
  dataset_id: number
  status: 'pending' | 'processing' | 'completed' | 'failed'
  total_count: number
  processed_count: number
  passed_count: number
  failed_count: number
  error_message?: string
  created_at: string
  updated_at: string
}

export interface BatchListResponse {
  items: Batch[]
  total: number
  skip: number
  limit: number
}

export interface BatchCreate {
  name: string
  description?: string
  rule_ids?: number[]
  rule_set_id?: number
  dataset_id: number
}

export interface BatchUpdate {
  name?: string
  description?: string
  rule_ids?: number[]
  status?: string
}

/**
 * 获取批次列表
 */
export function getBatchList(params: {
  skip?: number
  limit?: number
  search?: string
}): Promise<BatchListResponse> {
  return request.get('/batch', { params })
}

/**
 * 获取单个批次
 */
export function getBatch(id: number): Promise<Batch> {
  return request.get(`/batch/${id}`)
}

/**
 * 创建批次
 */
export function createBatch(data: BatchCreate): Promise<Batch> {
  return request.post('/batch', data)
}

/**
 * 更新批次
 */
export function updateBatch(id: number, data: BatchUpdate): Promise<Batch> {
  return request.put(`/batch/${id}`, data)
}

/**
 * 删除批次
 */
export function deleteBatch(id: number): Promise<void> {
  return request.delete(`/batch/${id}`)
}

/**
 * 执行批次质控
 */
export function executeBatch(id: number): Promise<Batch> {
  return request.post(`/batch/${id}/execute`)
}

/**
 * 规则执行结果
 */
export interface RuleExecutionResult {
  rule_id: number
  rule_name: string
  passed: boolean
  flag: number
  deduct: number
  explanation?: string
  answer?: any
  error?: string
  duration_ms: number
  created_at: string
}

/**
 * 数据项执行结果
 */
export interface DataItemExecutionResult {
  data_id: number
  file_name: string
  file_size: number
  overall_passed: boolean
  rule_results: RuleExecutionResult[]
  total_deduct: number
}

/**
 * 批次执行详情
 */
export interface BatchExecutionDetail {
  batch_id: number
  batch_name: string
  dataset_id: number
  dataset_name: string
  rule_ids: number[]
  rule_names: string[]
  status: string
  total_count: number
  processed_count: number
  passed_count: number
  failed_count: number
  data_results: DataItemExecutionResult[]
}

/**
 * 获取批次执行详情
 */
export function getBatchDetail(id: number): Promise<BatchExecutionDetail> {
  return request.get(`/batch/${id}/detail`)
}

File: backend/schemas/batch.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, Field, field_validator


class BatchBase(BaseModel):
    name: str = Field(..., description="批次名称")
    description: Optional[str] = Field(None, description="描述")
    rule_ids: List[int] = Field(default_factory=list, description="规则ID列表")
    rule_set_id: Optional[int] = Field(None, description="规则集ID（如果提供，将使用规则集中的规则）")
    dataset_id: int = Field(..., description="数据集ID")


class BatchCreate(BatchBase):
    pass


class BatchUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    rule_ids: Optional[List[int]] = None
    status: Optional[str] = None
    total_count: Optional[int] = None
    processed_count: Optional[int] = None
    passed_count: Optional[int] = None
    failed_count: Optional[int] = None
    error_message: Optional[str] = None


class BatchOut(BatchBase):
    id: int
    status: str
    total_count: int
    processed_count: int
    passed_count: int
    failed_count: int
    error_message: Optional[str] = None
    created_at: str
    updated_at: str

    @field_validator('created_at', 'updated_at', mode='before')
    @classmethod
    def format_datetime(cls, v):
        if isinstance(v, datetime):
            return v.strftime("%Y-%m-%d %H:%M:%S")
        return v


class BatchListResponse(BaseModel):
    items: List[BatchOut]
    total: int
    skip: int
    limit: int


class BatchExecuteRequest(BaseModel):
    """批次执行请求"""
    batch_id: int


class RuleExecutionResult(BaseModel):
    """规则执行结果"""
    rule_id: int
    rule_name: str
    passed: bool
    flag: int
    deduct: int
    explanation: Optional[str] = None
    answer: Optional[dict] = None
    error: Optional[str] = None
    duration_ms: int
    created_at: str


class DataItemExecutionResult(BaseModel):
    """数据项执行结果"""
    data_id: int
    file_name: str
    file_size: int
    overall_passed: bool  # 所有规则是否都通过
    rule_results: List[RuleExecutionResult]  # 每个规则的执行结果
    total_deduct: int  # 总扣分


class BatchExecutionDetail(BaseModel):
    """批次执行详情"""
    batch_id: int
    batch_name: str
    dataset_id: int
    dataset_name: str
    rule_ids: List[int]
    rule_names: List[str]
    status: str
    total_count: int
    processed_count: int
    passed_count: int
    failed_count: int
    data_results: List[DataItemExecutionResult]  # 每个数据项的执行结果
